dpnp/dpnp_iface_mathematical.py

```diff
@@ -1750,34 +1751,72 @@ def nancumsum(x1, **kwargs):
     return call_origin(numpy.nancumsum, x1, **kwargs)
 
 
-def nanprod(x1, **kwargs):
+def nanprod(
+    a,
+    axis=None,
+    dtype=None,
+    out=None,
+    keepdims=False,
+    initial=None,
+    where=True,
+):
     """
-    Calculate prod() function treating 'Not a Numbers' (NaN) as ones.
+    Return the product of array elements over a given axis treating Not a Numbers (NaNs) as ones.
 
     For full documentation refer to :obj:`numpy.nanprod`.
 
+    Returns
+    -------
+    out : dpnp.ndarray
+        A new array holding the result is returned unless `out` is specified, in which case it is returned.
+
+    See Also
+    --------
+    :obj:`dpnp.prod` : Returns product across array propagating NaNs.
+    :obj:`dpnp.isnan` : Test element-wise for NaN and return result as a boolean array.
+
     Limitations
     -----------
-    Parameter `x1` is supported as :obj:`dpnp.ndarray`.
-    Keyword argument `kwargs` is currently unsupported.
+    Input array is only supported as either :class:`dpnp.ndarray` or :class:`dpctl.tensor.usm_ndarray`.
+    Parameters `initial`, and `where` are only supported with their default values.
     Otherwise the function will be executed sequentially on CPU.
     Input array data types are limited by supported DPNP :ref:`Data types`.
 
     Examples
     --------
     >>> import dpnp as np
-    >>> np.nanprod(np.array([1, 2]))
-    2
-    >>> np.nanprod(np.array([[1, 2], [3, 4]]))
-    24
+    >>> np.nanprod(np.array(1))
+    array(1)
+    >>> np.nanprod(np.array([1]))
+    array(1)
+    >>> np.nanprod(np.array([1, np.nan]))
+    array(1.0)
+    >>> a = np.array([[1, 2], [3, np.nan]])
+    >>> np.nanprod(a)
+    array(6.0)
+    >>> np.nanprod(a, axis=0)
+    array([3., 2.])
+
+    """
+
+    if issubclass(a.dtype.type, dpnp.inexact):
```

Comment:
There might be no `dtype` attribute for input argument `a`, Looks as a check of `dpnp.is_supported_array_or_scalar(a)` is missing. Probably it would be good to validate a type of input array prior the check and to raise the `TypeError` exception if required.